pkg/controller/service/service_controller.go

```diff
@@ -264,7 +264,9 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 		needDelete := true
 		_, exists, err := s.balancer.GetLoadBalancer(s.clusterName, service)
 		if err != nil {
-			return fmt.Errorf("Error getting LB for service %s: %v", key, err), retryable
+			err = fmt.Errorf("Error getting LB for service %s: %v", key, err)
+			s.eventRecorder.Event(service, v1.EventTypeWarning,"GettingLoadBalancerFailed", err.Error())
```

Comment:
I think event recording should be handled at a more granulzr level than the creation func
Not sure if it counts as a duplicate as this contains more semantic and these are more for precision. If we really need to leep one I'd be in favour of granular events.